vault/identity_store_oidc.go

```diff
@@ -1683,10 +1744,20 @@ func (i *IdentityStore) oidcPeriodicFunc(ctx context.Context) {
 			if nextExpiration.Before(nextRun) {
 				nextRun = nextExpiration
 			}
+
+			if jwksClientCacheDuration < minJwksClientCacheDuration {
+				minJwksClientCacheDuration = jwksClientCacheDuration
+			}
 		}
 		if err := i.oidcCache.SetDefault(noNamespace, "nextRun", nextRun); err != nil {
 			i.Logger().Error("error setting oidc cache", "err", err)
 		}
+		if minJwksClientCacheDuration < math.MaxInt64 {
+			if err := i.oidcCache.SetDefault(noNamespace, "maxJwksClientCache", minJwksClientCacheDuration); err != nil {
```

Comment:
Suggest to rename `maxJwksClientCache` to something like `jwksClientCacheMaxAge` or `jwksCacheControlMaxAge` to make it more clear what the value is being used for. I was a bit confused why the min was being set to the max until I realized this was for the header :)